packages/terra-button-group/src/ButtonGroupButton.jsx

```diff
@@ -88,7 +88,7 @@ class ButtonGroupButton extends React.Component {
   handleKeyUp(event) {
     // Apply focus styles for keyboard navigation.
     // The onFocus event doesn't get triggered in some browsers, hence, the focus state needs to be managed here.
-    if (event.nativeEvent.keyCode === KeyCode.KEY_TAB || event.nativeEvent.keyCode === KeyCode.KEY_LEFT || event.nativeEvent.keyCode === KeyCode.KEY_RIGHT) {
+    if (event.keyCode === KeyCode.KEY_TAB || event.keyCode === KeyCode.KEY_LEFT || event.keyCode === KeyCode.KEY_RIGHT || event.keyCode === KeyCode.KEY_UP || event.keyCode === KeyCode.KEY_DOWN) {
```

Comment:
This seems to be a lot of conditions for a single if statement. For example, the keydown event will only be true after 4 failures. We can optimize this by using a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).

```js
const validKeys = new Set([key1, key2, ...]);

if(validKeys.has(event.keyCode))
   ....

```

Reply:
Updated condition using Set - [2696d6d](https://github.com/cerner/terra-core/pull/3857/commits/2696d6d45a49ed12b5690bdf2c1326b0e1a357a4)